ambry-store/src/main/java/com.github.ambry.store/Log.java

```diff
@@ -430,19 +430,17 @@ private void ensureCapacity(long writeSize) throws IOException {
    * @param segment the {@link LogSegment} instance to add.
    * @param increaseUsedSegmentCount {@code true} if the number of segments used has to be incremented, {@code false}
    *                                             otherwise.
-   * @throws IllegalArgumentException if the {@code segment} being added is past the active segment or if the segment
-   *                                  added cannot be counted as "used".
+   * @throws IllegalArgumentException if the {@code segment} being added is past the active segment
    */
   void addSegment(LogSegment segment, boolean increaseUsedSegmentCount) {
-    if (increaseUsedSegmentCount && remainingUnallocatedSegments.decrementAndGet() < 0) {
```

Comment:
During recovery, it is possible for it to go negative but it will never be negative in stable state.